operators/src/source/gdal_source.rs

```diff
@@ -820,771 +691,281 @@ mod tests {
-            time_step: TimeStep {
-                granularity: TimeGranularity::Seconds,
-                step: 1,
-            },
-        };
-
-        let gdal_params = GdalSourceParameters {
-            dataset_id: "modis_ndvi".to_owned(),
-            channel: None,
-        };
-
-        let dataset_information = JsonDatasetInformation {
-            time: time_interval_provider,
-            tile: None,
-            file_name_with_time_placeholder: "MOD13A2_M_NDVI_2014-01-01.TIFF".into(),
-            time_format: "".into(),
-            base_path: "../modis_ndvi".into(),
-            data_type: RasterDataType::U8,
-            geo_transform: dataset_geo_transform,
-            grid_shape: global_size_in_pixels.into(),
-            spatial_ref: SpatialReference::epsg_4326().into(),
-            measurement: Measurement::Unitless,
-        };
-
-        let dataset_information_provider = JsonDatasetInformationProvider {
-            dataset_information,
-            raster_data_root: "../operators/test-data/raster".into(),
-        };
-
-        let gdal_source = GdalSourceProcessor {
-            dataset_information: dataset_information_provider,
-            gdal_params,
-            tiling_specification: TilingSpecification {
-                origin_coordinate: Coordinate2D::new(0., 0.),
-                tile_size_in_pixels: GridShape2D::from(tile_size_in_pixels),
-            },
-            phantom_data: PhantomData,
-        };
+    fn test_load_tile_data_overlaps_dataset_bounds() {
+        let output_shape: GridShape2D = [8, 8].into();
+        let output_bounds = BoundingBox2D::new_unchecked((-360., 0.).into(), (0., 180.).into());
 
-        let vres: Vec<Result<RasterTile2D<u8>, Error>> = gdal_source
-            .time_tile_iter(
-                origin_split_tileing_strategy,
-                TimeInterval::new_unchecked(0, 1000),
-                bounding_box,
-            )
-            .map(|(time_interval, tile_information)| {
-                gdal_source.load_tile_data(time_interval, tile_information)
-            })
-            .collect();
-        assert_eq!(vres.len(), 6 * 3);
-        let upper_left_pixels: Vec<_> = vres
-            .into_iter()
-            .map(|t| {
-                let raster_tile = t.unwrap();
-                raster_tile
-                    .grid_array
-                    .get_at_grid_index([
-                        (tile_size_in_pixels[1] / 2) as isize,
-                        (tile_size_in_pixels[0] / 2) as isize,
-                    ])
-                    .unwrap() // pixel
-            })
-            .collect();
+        let x = load_ndvi_jan_2014(output_shape, output_bounds).unwrap();
 
-        let ndvi_center_pixel_values = vec![
-            19, 255, 255, 43, 76, 17, 255, 255, 255, 145, 255, 255, 255, 255, 255, 255, 255, 255,
-        ];
+        assert_eq!(x.data.len(), 64);
 
-        assert_eq!(upper_left_pixels, ndvi_center_pixel_values);
+        // TODO: compare actual data
```

Comment:
add test

Reply:
I am not sure how to verify the results. They are slightly differnent in QGIS but looks good to me 🤓

Reply:
well then we need a task to generate valid tests

Reply:
https://trello.com/c/nH4Rw9zK

Reply:
2fb88fa6eae13d7e3c6a90eedc19c8ff5dcc5d35